src/receiving.rs

```diff
@@ -90,19 +91,19 @@ impl From<Label> for Scalar {
 pub struct Receiver {
     version: u8,
     scan_privkey: SecretKey,
-    spend_privkey: SecretKey,
-    labels: HashMap<PublicKey, Label>,
+    spend_pubkey: PublicKey,
+    labels: BiMap<Label, PublicKey>,
```

Comment:
Ok so the point of Bimap here is to make sure that we can't have duplicate keys for the same value I think (2 different `Label` mapping to the same key) and to make effective reverse lookup (looking for a label by the pubkey).
It seems that the only time we need to do a reverse lookup is when generating an address for a label. 
So the trade off is having one more dependency + maintaining 2 maps internally (more memory consumption) to be faster when generating addresses for labels. I think it doesn't matter much in practice but still

Reply:
Yeah I think 2 hashmaps is better, I'll change this